chameleon-core/src/main/java/org/hibernate/omm/jdbc/MongoResultSet.java

```diff
@@ -103,144 +98,106 @@ public boolean wasNull() throws SimulatedSQLException {
         if (closed) {
             throw new ResultSetClosedSQLException();
         }
-        return lastRead != null && lastRead.isNull();
+        return requireNonNull(lastRead).isNull();
+    }
+
+    private BsonValue getBsonValue(int columnIndex) throws SimulatedSQLException {
+        beforeAccessCurrentDocumentField();
+        lastRead = castNonNull(currentDocument).get(getKey(columnIndex), BsonNull.VALUE);
```

Comment:
Helper method, default to `BsonNull.VALUE` so even if the document doesn't contain a value for the key at all, `lastRead` has a non-null value and a non-null value is returned from this method.

Reply:
btw, we have some perf opt trick here for we have used the returned doc field naming convention as following (from aggregate rendering we have done):
```
{ f0: ..., f1:..., f2:..., ... }
```
so we could simplify by replacing:
```
.get(getKey(columnIndex))
```
with
```
.get("f" + (columnIndex - 1))
```
but for now early-optimization might be a trap, just share the idea.